### PR TITLE
Enhance hooks tracker by adding an returned error to record function

### PR DESCRIPTION
**Pull request by allenxu404**
Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: internal/hook/wait_exec_hook_handler.go [hunk 1]
```diff
@@ -166,7 +166,11 @@ func (e *DefaultWaitExecHookHandler) HandleHooks(
 					err := fmt.Errorf("hook %s in container %s expired before executing", hook.HookName, hook.Hook.Container)
 					hookLog.Error(err)
 					errors = append(errors, err)
-					hookTracker.Record(newPod.Namespace, newPod.Name, hook.Hook.Container, hook.HookSource, hook.HookName, hookPhase(""), true)
+
+					trackerErr := hookTracker.Record(newPod.Namespace, newPod.Name, hook.Hook.Container, hook.HookSource, hook.HookName, hookPhase(""), true)
```
Comment (reasonerjt) [inline internal/hook/wait_exec_hook_handler.go:170]: Any reason you are ignoring the errors returned by `Record` in `internal/hook/item_hook_handler.go`
but handling the errors in this file?

File: internal/hook/item_hook_handler.go [hunk 2]
```diff
@@ -233,14 +233,14 @@ func (h *DefaultItemHookHandler) HandleHooks(
 			},
 		)
 
-		hookTracker.Record(namespace, name, hookFromAnnotations.Container, HookSourceAnnotation, "", phase, false)
-		if err := h.PodCommandExecutor.ExecutePodCommand(hookLog, obj.UnstructuredContent(), namespace, name, "<from-annotation>", hookFromAnnotations); err != nil {
+		hookFailed := false
+		if err = h.PodCommandExecutor.ExecutePodCommand(hookLog, obj.UnstructuredContent(), namespace, name, "<from-annotation>", hookFromAnnotations); err != nil {
```
Comment (reasonerjt) [inline internal/hook/item_hook_handler.go:237]: trivial: it'd be better to limit the scope of the var `err` to the `if` clause and use `hookFailed` in line 242

Comment (allenxu404) [reply internal/hook/item_hook_handler.go:237]: Sure, let me make it more clear.

File: internal/hook/wait_exec_hook_handler.go (hunk 1 above)
Comment (allenxu404) [reply internal/hook/wait_exec_hook_handler.go:170]: The `Record` error was ignored in `internal/hook/item_hook_handler.go` just for simplicity, which I consider might not be a good behavior. 

Will change it.

**Review by reasonerjt — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Pull request merged**